src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2267,6 +2271,81 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly referenceGenes = remoteData<ReferenceGenomeGene[]>({
+        await: ()=>[
+            this.studies
+        ],
+        invoke: () => {
+            return fetchAllReferenceGenomeGenes(this.studies.result[0].referenceGenome);
+        }
+    });
+
+    readonly hugoGeneSymbolToReferenceGene = remoteData({
+        await: ()=>[
+            this.referenceGenes
+        ],
+        invoke: ()=>{
+            // build reference gene map
+            const result:{[hugosymbol:string]:ReferenceGenomeGene} =
+                _.keyBy(this.referenceGenes.result!, g=>g.hugoGeneSymbol);
+            return Promise.resolve(result);
+        }
+    });
+
+    readonly mutationEnrichmentData  = remoteData({
+        await: ()=>[
+            this._mutationEnrichmentDataWithoutCytoband,
```

Comment:
since you're now just doing `fetchAllReferenceGenomeGenes`, theres no need to do this thing anymore where we calculate the cytoband afterwards. see what I mean? so you can just put it back to the way you had it before, where you add the cytoband inside makeEnrichmentsDataPromise